Linux/sgx/libsgx_tsgxssl/Makefile

```diff
@@ -93,6 +93,12 @@ Sgx_tssl_Include_Paths := -I. -I$(PACKAGE_INC) -I$(SGX_SDK_INC) -I$(SGX_SDK_INC)
 Common_C_Cpp_Flags := -DOS_ID=$(OS_ID) $(SGX_COMMON_CFLAGS) -nostdinc -fdata-sections -ffunction-sections -Os -Wl,--gc-sections -fvisibility=hidden -fpie -fpic -fstack-protector -fno-builtin-printf -Wformat -Wformat-security $(Sgx_tssl_Include_Paths)
 Sgx_tssl_C_Flags := $(Common_C_Cpp_Flags) -Wno-implicit-function-declaration -std=c11 $(MITIGATION_CFLAGS) $(NO_THREADS_CFLAG)
 Sgx_tssl_Cpp_Flags := $(Common_C_Cpp_Flags) -std=c++11 -nostdinc++ $(MITIGATION_CFLAGS)
+ifeq ($(SGX_MODE), SIM)
+ifeq ($(NONE_INTEL),TRUE)
+Sgx_tssl_C_Flags += -DSKIP_INTELCPU_CHECK
+endif
+endif
```

Comment:
As far as I understand, building SGX SSL should be independent of the mode the user application enclave uses. One shouldn't be required to have two different versions of the SGX SSL library. It's up to the downstream application to decide the mode, e.g., the test app in this repository.

Either remove the check for the `SGX_MODE` or remove the check for an Intel CPU altogether (the SDK doesn't have this check, why would SGX SSL have it?).

Reply:
Just curious if there is a non-Intel CPU which is capable of SGX Hardware mode.

Reply:
There are none, I guess. Nevertheless, it would make packaging easier if there is no distinction between simulation and HW. Otherwise, depending on the build mode, an enclave would require different builds of SGX SSL, making the dependency management unnecessarily harder.

Reply:
Simulation mode is NOT a proper option for making packaging. It is only used to simulate a SGX SSL project on any Intel platform. Removing it will force developers to debug the sample project on a SGX supported platform. I don't think it a good idea.

I am thinking about making `NONE_INTEL=TRUE SGX_MODE=SIM` into `SGX_MODE=SIM`, and skip checking CPU type in this mode, since it was mentioned that https://github.com/intel/intel-sgx-ssl/issues/113#issuecomment-1022953038.

Reply:
Implemented. Now the usage is changed to
`Linux$ SGX_MODE=SIM make`

Reply:
I think we have a misunderstanding here. @Trundle was talking about the packaging of `intel-sgx-ssl`, not the final enclave application which makes use of `intel-sgx-ssl`.

Consider the Intel `linux-sgx` SDK: Intel offers various prebuilt packages, e.g., for Ubuntu/Debian or Red Hat/Centos. Developers use those packages to develop enclave applications. To that end, they use `SGX_MODE=SIM` during development and `SGX_MODE=HW` for the enclave application deployed to production. That means, the Intel `linux-sgx` SDK leaves the mode of operation to the application; developers don't have to install an `SW` and an `HW` flavor of the Intel `linux-sgx` SDK.

This is very different from the solution you are currently implementing: if one wants to build an enclave application using `intel-sgx-ssl` with `SGX_MODE=SIM` they're now required to first install/build a simulation version of `intel-sgx-ssl`; likewise, if they want to build using `SGX_MODE=HW` they have to install/build a hardware version of `intel-sgx-ssl`. This shouldn't be required: enclaves built with `SGX_MODE=SIM` link against the simulation libraries, enclaves built with `SGX_MODE=HW` link against the hardware libraries.

Reply:
In my opinion, you don't have to do packing of intel-sgx-ssl by yourself. The [make file](https://github.com/intel/intel-sgx-ssl/blob/master/Linux/Makefile) here is only for the final enclave application.

There is actually another way to integrate Intel SGX SSL in the SGX SDK for Linux.  As described in:  
https://github.com/intel/linux-sgx/blob/master/README.md#build-the-intelr-sgx-sdk-and-intelr-sgx-sdk-installer . 
If you choose `USE_OPT_LIBS=0` or `USE_OPT_LIBS=2` in building SGX SDK, the generated `libsgx_tcrypto.a` will cover all the functions of `libsgx_tsgxssl_crypto.a` and `libsgx_tsgxssl.a`. There should be no problem unless you also need the crypto library from IPP.

Reply:
Supporting non-Intel CPU with SGX hardware mode was not in the SGX SSL design scope. There will be unknown results for that because it has never been validated.

Reply:
I agree that we don't need to use SGX_MODE. Directly using SKIP_INTELCPU_CHECK is enough.

Reply:
SKIP_INTELCPU_CHECK=TRUE is now the only flag to satisfy the requirement.